design/collision.layers.v2.md

```diff
@@ -0,0 +1,60 @@
+## One Line Concept:
+Change collision layer system to be more intuitive and safely support group masks and exclusive actors, UI, and triggers
+
+## Problem Description: 
+The collision system is used in a number of different cases, such as: Rigid Body Collision with Kinematic/Static objects or with other rigid bodies, Player Controller collision, Teleport ray collisions, UI ray collisions, proximity grab, Trigger Zones (for rigid bodies and for players.)
+
+The current collision layer implementation loosely wraps the Unity collision layers, and allowing 4 different layers (Default, NavMesh, Hologram, UI) to be assigned freely to all colliders regardless of usage.
+
+The design had to minimize number of collision layers (as that is a shared resource in Unity, and there are only 32 layers shared between all collision and all rendering), and that affected the design.
+
+The current implementation is problematic, because
+1. it's not intuitive what the different types do, or when they should be used.
+2. if using group masked/exclusive actors in certain cases, they can cause issues with rigid body simulation not being (largely) globally consistent across all connected clients
```

Comment:
I do not see why this should be an issue. I think the collision layers are syncronized across all clients then even they are not globally consistent but they always should tend to the same global state. 
Or is it meant here that the prediction code does not support (yet) the different collision layers (?)